cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -376,6 +379,9 @@ public OptimizerResult optimizations(ClusterModel clusterModel,
    * @param isTriggeredByGoalViolation True if optimization of goals is triggered by goal violation, false otherwise.
    * @param requestedDestinationBrokerIds Explicitly requested destination broker Ids to limit the replica movement to
    *                                      these brokers (if empty, no explicit filter is enforced -- cannot be null).
+   * @param initReplicaDistributionForProposalGeneration The initial replica distribution of the cluster.
+   * @param initLeaderDistributionForProposalGeneration The initial leader replica distribution of the cluster.
```

Comment:
`initReplicaDistributionForProposalGeneration` and `initLeaderDistributionForProposalGeneration` need more explanation in JavaDoc. It should answer questions such as when is it relevant and what its value should be when it is not relevant (i.e. `null`)